example/speedtest/sub/main.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) TrueBrain
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <TrueMQTT.h>
+#include <chrono>
+#include <iostream>
+#include <thread>
+#include <unistd.h>
+#include <vector>
+
+
+enum TIME_RES
+{
+    T_MICRO,
+    T_MILLI,
+    T_SECONDS
+};
+
+std::vector<int> vecTimings;
+
+std::uint64_t getEpochUSecs()
+{
+    auto tsUSecs = std::chrono::time_point_cast<std::chrono::microseconds>(std::chrono::system_clock::now());
+    return static_cast<std::uint64_t>(tsUSecs.time_since_epoch().count());
+}
+
+void processTimes(int epochTat)
+{
+    vecTimings.emplace_back( epochTat );
+
+    if(vecTimings.size() == 10)
+    {
+        float avgTat = 0;
+        int totalTat = 0;
+        for(auto timing : vecTimings)
+        {
+            totalTat += timing;
+        }
+        avgTat = totalTat / vecTimings.size();
```

Comment:
Why use a vector here, and not just 2 variables:

`totalTiming` and `count`.

As it seems that is all you are doing here, adding up all timings and dividing by the total?

Reply:
To be able to get as much messages in the shortest time, I chose to put everything in a vector and calculate the average per 10 messages. Just another way to do it, I guess to keep the timings as close to the truth as possible.